src/Reflection/ReflectionClass.php

```diff
@@ -1618,6 +1619,53 @@ private function getInterfacesHierarchy(array &$interfaceClassNames = []): array
         return $this->addStringableInterface($interfaces);
     }
 
+    /** @return list<ReflectionClass> */
+    private function getCurrentClassUsedTraits(): array
+    {
+        if ($this->isTrait) {
+            // assumption: first key is the current trait
+            return array_slice($this->getTraitsHierarchy(), 1);
+        }
+
+        return array_merge(
+            [],
+            ...array_map(
+                static fn (ReflectionClass $trait): array => $trait->getTraitsHierarchy(),
+                $this->getTraits(),
+            ),
+        );
+    }
+
+    /**
+     * This method allows us to retrieve all traits used by this trait. Do not use on class nodes!
+     *
+     * @param list<trait-string> $traitClassNames
+     *
+     * @return list<ReflectionClass>
+     */
+    private function getTraitsHierarchy(array &$traitClassNames = []): array
```

Comment:
ah, this name is also weird. it's not a hierarchy. maybe more like `getFlattenedTraits` or so? The only difference between this and the other new method is then only that one works only on traits and includes the current trait and the other works on everything and does not include the current trait 🤔